base/src/main/java/uk/ac/ebi/atlas/search/analyticsindex/FullAnalyticsSearchService.java

```diff
@@ -0,0 +1,251 @@
+    // private static final String NON_SELECTED_ASSAY_GROUP_IDS_Q_TEMPLATE = "('{'!terms f=bioentity_identifier'}'{0}) AND expression_level:[{1} TO *]";
+    // private static final String NON_SELECTED_ASSAY_GROUP_IDS_Q_TEMPLATE_ALT = "expression_level:[{1} TO *]";
+
+    private static final String SELECTED_ASSAY_GROUP_IDS_Q_TEMPLATE = "('{'!terms f=assay_group_id'}'{0}) AND expression_level:[{1} TO *]";
+    private static final String NON_SELECTED_ASSAY_GROUP_IDS_Q_TEMPLATE = "-('{'!terms f=assay_group_id'}'{0}) AND ('{'!terms f=bioentity_identifier'}'{1}) AND expression_level:[{2} TO *]";
+    private static final String NON_SELECTED_ASSAY_GROUP_IDS_Q_TEMPLATE_ALT = "-('{'!terms f=assay_group_id'}'{0}) AND expression_level:[{1} TO *]";
+
+
+    private final RestTemplate restTemplate;
+    private final Resource avgExpressionOverBioentityIdentifiersFacet;
+    private final Resource maxExpressionOverBioentityIdentifiersFacet;
+
+    @Inject
+    public FullAnalyticsSearchService(RestTemplate restTemplate,
+                                      @Value("classpath:/uk/ac/ebi/atlas/search/baseline/solr/avgExpressionOverBioentityIdentifiers.json")
+                             Resource avgExpressionOverBioentityIdentifiersFacet,
+                                      @Value("classpath:/uk/ac/ebi/atlas/search/baseline/solr/maxExpressionOverBioentityIdentifiers.json")
+                             Resource maxExpressionOverBioentityIdentifiersFacet) throws IOException {
+        this.restTemplate = restTemplate;
+        this.avgExpressionOverBioentityIdentifiersFacet = avgExpressionOverBioentityIdentifiersFacet;
+        this.maxExpressionOverBioentityIdentifiersFacet = maxExpressionOverBioentityIdentifiersFacet;
+    }
+
+    public List<String> searchSpecificExpression(String experimentAccession,
+                                                 Double expressionThreshold,
+                                                 Set<String> assayGroupIds) {
+        return searchSpecificExpression(experimentAccession, expressionThreshold, assayGroupIds, DEFAULT_RESULT_SIZE);
+    }
+
+    public List<String> searchSpecificExpression(String experimentAccession,
+                                                 Double expressionThreshold,
+                                                 Set<String> assayGroupIds,
+                                                 int resultSize) {
+
+        String filterQueries = MessageFormat.format(FQ_TEMPLATE, experimentAccession);
+
+        SolrQuery solrQuery = new SolrQuery();
+        solrQuery.setRows(0)
+                .setFilterQueries(filterQueries)
+                .setQuery(
+                        MessageFormat.format(SELECTED_ASSAY_GROUP_IDS_Q_TEMPLATE, Joiner.on(",").join(assayGroupIds), expressionThreshold))
+                .set("json.facet", ResourceUtils.readPlainTextResource(avgExpressionOverBioentityIdentifiersFacet));
```

Comment:
Perhaps: do `ResourceUtils.readPlainTextResource` at startup. Make countless scientists wait for their results a small bit less.